### PR TITLE
fix(cmf): remove automatic event from onEvent [BC]

### PR DESCRIPTION
**Pull request by sgendre**
**What is the problem this PR is trying to solve?**
The SyntheticEvent that is passed can be quite fat, leading devTools to crash by oom or **browser to lag**

a static object  to define the action to provide to a button
```javascript

const submitButtonProps = {
	type: 'submit',
	label: 'Select that',
	onClickDispatch: { type: SUBMIT_ACTION },
};
```

**What is the chosen solution to this problem?**
If the event is a react SyntheticEvent, provide an empty object instead of nested object with potential circular dependencies

if you need data from the event initial, implement your own handler and extract your data to build your own minimum meaningful payload
```javascript
const submitButtonProps = dispatch => {
	const dispatchSubmit = e =>
		dispatch({ type: SUBMIT_ACTION, payload: { target: e.target } });
	return {
		type: 'submit',
		label: 'Select that',
		onClick: dispatchSubmit,
	};
};
```

Migration path:

```snippet
// before
const submitButtonProps = {
	type: 'submit',
	label: 'Select that',
	onClickDispatch: { type: SUBMIT_ACTION },
};

// after
const submitButtonProps = dispatch => {
	const dispatchSubmit = e =>
		dispatch({ type: SUBMIT_ACTION, payload: { target: e.target } });
	return {
		type: 'submit',
		label: 'Select that',
		onClick: dispatchSubmit,
	};
};
```

alternative solution can be found [here](https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/Troubleshooting.md#it-fails-to-serialize-data-when-passing-synthetic-events-or-calling-an-action-directly-with-redux-actions)
**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[x] This PR introduces a breaking change**

If you look for the original native event, you will only get an empty object.
The best is to catch that native event yourself and dispatch your own event with the minimal payload.

### CODE REVIEW

File: packages/cmf/src/onEvent.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ import CONSTANT from './constant';
 function serializeEvent(event) {
 	if (event.persist) {
 		event.persist();
+		return {};
```
Comment (jsomsanith) [inline packages/cmf/src/onEvent.js]: If the event is not used/propagated, why do you still persist it ?

Comment (sgendre) [reply packages/cmf/src/onEvent.js]: I bet that some middleware or js-logger could have some reference on it?

Comment (jmfrancois) [reply packages/cmf/src/onEvent.js]: No you can t have any reference to it if you don t dispatch it.

Comment (sgendre) [reply packages/cmf/src/onEvent.js]: k thx, updated

**Pull request merged**